libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateTimePicker.tsx

```diff
@@ -0,0 +1,164 @@
+import { injectIntl, WrappedComponentProps } from "react-intl";
+import { IntlWrapper } from "@gooddata/sdk-ui";
+import DayPickerInput from "react-day-picker/DayPickerInput";
+import moment from "moment";
+import { DateRangePickerInputFieldBody } from "./DateRangePickerInputFieldBody";
+import { convertPlatformDateStringToDate } from "../utils/DateConversions";
+import { TIME_FORMAT } from "../constants/Platform";
+
+interface IDateTimePickerOwnProps {
+    placeholderDate: string;
+    dateFormat: string;
+    onChange: (value: Date) => any;
+    value: Date;
+    dayPickerPropsWithDefaults: any;
+    handleDayClick: any;
+    locale: string;
+    isMobile: boolean;
+    isTimeForStaticPeriodEnabled: boolean;
+    className: string;
+    defaultTime?: string;
+    error?: boolean;
+}
+
+export type DateTimePickerComponentProps = IDateTimePickerOwnProps & WrappedComponentProps;
+
+// eslint-disable-next-line react/display-name
+export const DateTimePickerComponent = React.forwardRef<DayPickerInput, DateTimePickerComponentProps>(
+    (props: DateTimePickerComponentProps, ref) => {
+        const {
+            placeholderDate,
+            value,
+            onChange,
+            dateFormat,
+            dayPickerPropsWithDefaults,
+            handleDayClick,
+            isMobile,
+            isTimeForStaticPeriodEnabled,
+            className,
+            defaultTime = "00:00",
+            error = false,
+        } = props;
+
+        const getFormattedDate = () => {
+            return moment(value).format("YYYY-MM-DD");
+        };
+
+        const getFormattedTime = () => {
+            if (typeof value !== ("undefined" || null)) {
+                return moment(value).format(TIME_FORMAT);
+            }
+            return moment(null).format(TIME_FORMAT);
+        };
+
+        const onDateChange = (selectedDate: Date) => {
+            if (isTimeForStaticPeriodEnabled && selectedDate) {
+                let oldDate = value;
+                if (!oldDate) {
+                    oldDate = moment(defaultTime).toDate();
+                }
+
+                let updatedDatetime = moment(selectedDate).hours(oldDate.getHours());
+                updatedDatetime = moment(updatedDatetime).minutes(oldDate.getMinutes());
+
+                // set default time if necessary && configured
+                if (updatedDatetime.isSame(selectedDate)) {
+                    const time = moment(defaultTime, TIME_FORMAT).toDate();
+                    updatedDatetime.hours(time.getHours());
+                    updatedDatetime.minutes(time.getMinutes());
+                }
+
+                selectedDate = updatedDatetime.toDate();
+            }
+
+            onChange(selectedDate);
+        };
+
+        const onTimeChange = (input: string) => {
+            let date = value;
+            if (!date) {
+                // set today in case of invalid date
+                date = new Date();
+            }
```

Comment:
Again, all of the above can be written on a single line. See the possible way how to do it in similar comment above.

Reply:
````
const date = value ?? new Date();
const time = moment(input, TIME_FORMAT);
if (time.isValid()) {
    date.setHours(time.hours());
    date.setMinutes(time.minutes());
 }
onChange(date);
````

I must say that I really hate the mutability aspect of `moment` but that is probably personal pet peeve.

Reply:
fixed